packages/discovery-provider/src/queries/query_helpers.py

```diff
@@ -701,6 +712,12 @@ def _populate_gated_content_metadata(session, entities, current_user_id):
                     "stream": True,
                     "download": False,
                 }
+                if entity.get("track_id"):
+                    entity[
+                        response_name_constants.stream
+                    ] = get_stream_url_with_mirrors(
+                        entity, current_user_id, is_authorized_as_user=False
+                    )
         return
```

Comment:
early return here was causing this data to be missing

Reply:
Nice catch... don't know how I missed this when testing - could have sworn I tried this case...

Was the fallback to `stream` endpoint not working??

Reply:
seemingly not working correctly on iOS safari because of the way 302 is handled. this isn't a root cause fix, but something i saw to make life better... i don't know the root cause still :(